ubuntu/java/openjdk-8/Dockerfile

```diff
@@ -96,42 +96,64 @@ COPY dockerd-entrypoint.sh /usr/local/bin/
-    && apt-get -y install openjdk-$JAVA_VERSION-jdk=$JDK_VERSION \
+    && apt-get install -y python-setuptools=$PYTHON_TOOL_VERSION \
+
+    # Install OpenJDK 8
+    && apt-get install -y openjdk-${JAVA_VERSION}-jdk=$JDK_VERSION \
+    && apt-get install -y --no-install-recommends ca-certificates-java \
     && apt-get clean \
     # Ensure Java cacerts symlink points to valid location
     && update-ca-certificates -f \
-    && mkdir -p /usr/src/ant \
-    && wget "http://archive.apache.org/dist/ant/binaries/apache-ant-$ANT_VERSION-bin.tar.gz" -O /usr/src/ant/apache-ant-$ANT_VERSION-bin.tar.gz \
-    && tar -xzf /usr/src/ant/apache-ant-$ANT_VERSION-bin.tar.gz -C /usr/local \
-    && ln -s /usr/local/apache-ant-$ANT_VERSION/bin/ant /usr/bin/ant \
-    && rm -rf /usr/src/ant \
-    && mkdir -p /usr/share/maven /usr/share/maven/ref $MAVEN_CONFIG \
-    && curl -fsSL "https://archive.apache.org/dist/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz" \
-        | tar -xzC /usr/share/maven --strip-components=1 \
-    && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn \
-    && mkdir -p /usr/src/gradle \
-    && wget "https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip" -O /usr/src/gradle/gradle-$GRADLE_VERSION-bin.zip \
-    && unzip /usr/src/gradle/gradle-$GRADLE_VERSION-bin.zip -d /usr/local \
-    && ln -s /usr/local/gradle-$GRADLE_VERSION/bin/gradle /usr/bin/gradle \
-    && rm -rf /usr/src/gradle \
-    && rm -fr /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+    # Install Ant
+    && curl -LSso /var/tmp/apache-ant-$ANT_VERSION-bin.tar.gz https://archive.apache.org/dist/ant/binaries/apache-ant-$ANT_VERSION-bin.tar.gz  \
+    && echo "$ANT_DOWNLOAD_SHA512 /var/tmp/apache-ant-$ANT_VERSION-bin.tar.gz" | sha512sum -c - \
+    && tar -xzf /var/tmp/apache-ant-$ANT_VERSION-bin.tar.gz -C /opt \
+    && update-alternatives --install /usr/bin/ant ant /opt/apache-ant-$ANT_VERSION/bin/ant 10000 \
+
+    # Install Maven
+    && mkdir -p $MAVEN_HOME \
+    && curl -LSso /var/tmp/apache-maven-$MAVEN_VERSION-bin.tar.gz https://apache.org/dist/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz \
+    && echo "$MAVEN_DOWNLOAD_SHA1 /var/tmp/apache-maven-$MAVEN_VERSION-bin.tar.gz" | sha1sum -c - \
+    && tar xzvf /var/tmp/apache-maven-$MAVEN_VERSION-bin.tar.gz -C $MAVEN_HOME --strip-components=1 \
+    && update-alternatives --install /usr/bin/mvn mvn /opt/maven/bin/mvn 10000 \
+    && mkdir -p $MAVEN_CONFIG \
+
+    # Install Gradle
+    && curl -LSso /var/tmp/gradle-$GRADLE_VERSION-bin.zip https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip \
+    && echo "$GRADLE_DOWNLOAD_SHA256 /var/tmp/gradle-$GRADLE_VERSION-bin.zip" | sha256sum -c - \
+    && unzip /var/tmp/gradle-$GRADLE_VERSION-bin.zip -d /opt \
+    && update-alternatives --install /usr/local/bin/gradle gradle /opt/gradle-$GRADLE_VERSION/bin/gradle 10000 \
+
+    # Install SBT
```

Comment:
I think I'd drop SBT unless it's a strict requirement. Customers previously haven't been getting SBT so anyone inheriting from latest this image and installing SBT will run into problems. Beyond that adding SBT is a functionality change in what otherwise is a basic version bump CR. Best practice would be to keep those separate.

Reply:
I'm just now noticing it got added. I agree.